client/question/services/FeedbackGeneratorServiceSpec.js

```diff
@@ -1384,4 +1384,61 @@ describe('FeedbackGeneratorService', function() {
       }
     );
   });
+
+  describe('prereqCheck errorLineNumber', function() {
+    it('should return null when there are no errors', function() {
+      var starterCode = [
+        'def myFunction(arg):',
+        '    return arg',
+        ''
+      ].join('\n');
+
+      var prereqFailure = PrereqCheckFailureObjectFactory.create(
+        PREREQ_CHECK_TYPE_WRONG_LANG, null, starterCode, 'notOp');
+      var feedback = FeedbackGeneratorService.getPrereqFailureFeedback(
+        prereqFailure);
+
+      expect(feedback.getErrorLineNumber()).toBeNull();
+    });
+
+    it('should return append the errorStr', function() {
+      var incrementErrorCode = [
+        'def myFunction(arg):',
+        '    arg++',
+        '    return arg',
+        ''
+      ].join('\n');
+
+      var prereqFailure = PrereqCheckFailureObjectFactory.create(
+        PREREQ_CHECK_TYPE_WRONG_LANG, null, incrementErrorCode, 'incrementOp',
+        2, 4);
+      var feedback = FeedbackGeneratorService.getPrereqFailureFeedback(
+        prereqFailure);
+      var paragraphs = feedback.getParagraphs();
+
+      expect(feedback.getErrorLineNumber()).toBe(2);
+      expect(paragraphs[1].getContent()).toBe('(See line 2 of the code.)');
+    });
+
+    it('only return first errorLineNumber when multiple errors', function() {
```

Comment:
This test should be broader, since the code is not run -- so what this test is really testing is that if you supply line number 3 then the feedback will say line 3, which is unsurprising.

Instead, I think you can have this particular test as one of the end-to-end tests in SolutionHandlerServiceSpec.js. Just submit some code that fails more than one prerequisite check and check what feedback comes out.

Reply:
Done.